core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2193,6 +2195,78 @@ class WorkspaceService(
+      .getResourceAuthDomain(SamResourceTypeNames.workspace, workspace.workspaceId, parentContext)
+
+    _ <-
+      if ((newAuthDomainGroups -- existingAuthDomain).nonEmpty) {
+        samDAO.addResourceAuthDomain(SamResourceTypeNames.workspace,
+                                     workspace.workspaceId,
+                                     newAuthDomainGroups,
+                                     parentContext
+        )
+      } else {
+        // if the new auth domain is already in the workspace, do nothing
+        Future.successful(())
+      }
+    _ <-
+      if (existingAuthDomain.isEmpty) {
+        // if there was no auth domain, we need update the google bucket IAM to use the workspace's project owner policy
+        // instead of the billing project owner policy directly
+        changeProjectOwnerBucketIamBinding(workspace, parentContext)
+      } else {
+        Future.successful(())
+      }
+  } yield ()
+
+  private def changeProjectOwnerBucketIamBinding(workspace: Workspace, parentContext: RawlsRequestContext) =
+    for {
+      workspacePolicies <- samDAO.listPoliciesForResource(SamResourceTypeNames.workspace,
+                                                          workspace.workspaceId,
+                                                          parentContext
+      )
+      _ <- samDAO.syncPolicyToGoogle(SamResourceTypeNames.workspace,
+                                     workspace.workspaceId,
+                                     SamWorkspacePolicyNames.projectOwner
+      )
+      workspaceProjectOwnerEmail = workspacePolicies
+        .find(_.policyName == SamWorkspacePolicyNames.projectOwner)
+        .map(_.email)
+        .getOrElse(
+          throw new RawlsExceptionWithErrorReport(
+            ErrorReport(StatusCodes.InternalServerError,
+                        s"Unable to find project owner policy for workspace ${workspace.workspaceId}"
+            )
+          )
+        )
+      billingProjectPolicies <- samDAO.listPoliciesForResource(SamResourceTypeNames.billingProject,
+                                                               workspace.namespace,
+                                                               parentContext
+      )
+      billingProjectOwnerEmail = billingProjectPolicies
+        .find(_.policyName == SamBillingProjectPolicyNames.owner)
+        .map(_.email)
+        .getOrElse(
+          throw new RawlsExceptionWithErrorReport(
+            ErrorReport(StatusCodes.InternalServerError,
+                        s"Unable to find project owner policy for billing project ${workspace.namespace}"
+            )
+          )
+        )
+      _ <- gcsDAO.changeProjectOwnerBucketIamBinding(GcsBucketName(workspace.bucketName),
+                                                     Identity.group(billingProjectOwnerEmail.value),
+                                                     Identity.group(workspaceProjectOwnerEmail.value)
+      )
+    } yield ()
```

Comment:
What state will things be left in if the earlier call to `samDAO.addResourceAuthDomain` succeeds, but something fails before this executes `gcsDAO.changeProjectOwnerBucketIamBinding`? I don't know of a simple way to handle that case - perhaps worth logging a stern error so we might at least get a Sentry notification?

Reply:
updated code, see comments